src/java/org/apache/cassandra/db/commitlog/CommitLogSegment.java

```diff
@@ -175,7 +177,10 @@ static long getNextId()
 
         try
         {
-            channel = FileChannel.open(logFile.toPath(), StandardOpenOption.WRITE, StandardOpenOption.READ, StandardOpenOption.CREATE);
+            if(commitLog.configuration.isDirectIOEnabled())
+                channel = FileChannel.open(logFile.toPath(), StandardOpenOption.WRITE, StandardOpenOption.READ, StandardOpenOption.CREATE, ExtendedOpenOption.DIRECT);
```

Comment:
seems have not make an implementation for commitlog reader ?

this patch only work for write path , right ?

Reply:
Yes. Reading not done yet.

Reply:
as I have said before ，what about add a log for read ? if commitlog_disk_access_mode is direct_io , and for read here, we can add a warn log ,and tell the user, the direct io for read is not supported now, use the default or other instand.

Reply:
Agree and will add a warn log for read case.